[8.0] [ADD] l10n_es_igic

daviddiz opened the pull request:
IGIC (Impuesto General Indirecto Canario)

```
Define plantillas de impuestos IGIC.
Define plantillas de códigos de impuestos IGIC.
Define plantillas de cuentas contables IGIC.
```

daviddiz committed:
Módulo de IGIC

rafaelbn commented:
@daviddiz you need to sign CLA, please follow above instructions

daviddiz committed:
remove blank lines

acysos commented:
Instala y funciona sin problemas. Como ha comentado Rafael, faltaría firmar el CLA de OCA, para poder incluirlo.

Saludos

pedrobaeza commented:
Veo varios problemas en este PR:
- Los códigos de impuestos utilizan la nomenclatura antigua, declarando códigos como `--`, o  con los números de casilla que pueden ir variando con el tiempo. Eso hace no mantenible ni actualizable el plan en el futuro. Lo que se hizo en l10n_es es poner el nº de casilla como parte de la descripción, y dejar un código fijo único (se puede hacer a partir de las siglas de la descripción). Ejemplo: https://github.com/OCA/l10n-spain/blob/8.0/l10n_es/data/tax_codes_common.xml#L64
- Aunque no es obligatorio, separar cada tipo de dato (cuenta, impuesto, código de impuesto) en un archivo ayuda al mantenimiento.
- No hay que crear una cuenta de impuesto para cada porcentaje (cuentas 472, 477...). Esto fue un error muy grande de la versión 6.1 y anteriores que provocó algunos problemas precisamente con el IGIC al colisionar varias cuentas con el mismo número que se declaraban varias veces. Además, es totalmente innecesario en Odoo. Son los contales añejos los que se aferran a desglosar cuentas de impuestos así como desglosar las cuentas de clientes y proveedores.
- No se deben utilizar los mismos planes contables ya existentes para el IGIC, porque si se tiene una multi-compañía, y sólo una de ellas es de Canarias, a partir de ese momento cuando actualices el plan contable todas las compañías tendrán los impuestos IGIC y todos esos datos serán incorrectos en el resto de compañías. Lo suyo es crear un(os) plan(es) contable(s) nuevo(s) que derive de los anteriores (o del padre), y declaren esos impuestos en él. Gracias a la posibilidad de árbol jerárquico en los planes contables, es posible hacerlo sin tener que definir muchas más cosas.
- Habría que cambiar las posiciones fiscales, ya que están centradas en una empresa de la península, y algunos conceptos de Régimen intracomunitario, extracomunitario, etc, no son correctos para Canarias.

syci commented:
Confirmo lo que dice Pedro, yo también me vi con el solape de la cuentas existentes con la 6.1 y corregí con Pedro. Creo que lo mejor es incorporar la parte de Canarias en el Plan General en los XML base para no tener módulos separados.

> El 28 oct 2015, a las 18:53, Pedro M. Baeza notifications@github.com escribió:
> 
> Veo varios problemas en este PR:
> 
> Los códigos de impuestos utilizan la nomenclatura antigua, declarando códigos como --, o con los números de casilla que pueden ir variando con el tiempo. Eso hace no mantenible ni actualizable el plan en el futuro. Lo que se hizo en l10n_es es poner el nº de casilla como parte de la descripción, y dejar un código fijo único (se puede hacer a partir de las siglas de la descripción). Ejemplo: https://github.com/OCA/l10n-spain/blob/8.0/l10n_es/data/tax_codes_common.xml#L64 https://github.com/OCA/l10n-spain/blob/8.0/l10n_es/data/tax_codes_common.xml#L64
> Aunque no es obligatorio, separar cada tipo de dato (cuenta, impuesto, código de impuesto) en un archivo ayuda al mantenimiento.
> No hay que crear una cuenta de impuesto para cada porcentaje (cuentas 472, 477...). Esto fue un error muy grande de la versión 6.1 y anteriores que provocó algunos problemas precisamente con el IGIC al colisionar varias cuentas con el mismo número que se declaraban varias veces. Además, es totalmente innecesario en Odoo. Son los contales añejos los que se aferran a desglosar cuentas de impuestos así como desglosar las cuentas de clientes y proveedores.
> No se deben utilizar los mismos planes contables ya existentes para el IGIC, porque si se tiene una multi-compañía, y sólo una de ellas es de Canarias, a partir de ese momento cuando actualices el plan contable todas las compañías tendrán los impuestos IGIC y todos esos datos serán incorrectos en el resto de compañías. Lo suyo es crear un(os) plan(es) contable(s) nuevo(s) que derive de los anteriores (o del padre), y declaren esos impuestos en él. Gracias a la posibilidad de árbol jerárquico en los planes contables, es posible hacerlo sin tener que definir muchas más cosas.
> Habría que cambiar las posiciones fiscales, ya que están centradas en una empresa de la península, y algunos conceptos de Régimen intracomunitario, extracomunitario, etc, no son correctos para Canarias.
> —
> Reply to this email directly or view it on GitHub https://github.com/OCA/l10n-spain/pull/235#issuecomment-151953664.

pedrobaeza commented:
No, Mario, no he dicho de juntarlo en el _l10n_es_, si no de crear un objeto plan contable distinto, que herede del ya existente, pero que permita seleccionar "PGCE PYMEs" y "PGCE PYMEs Canarias" por separado. Tal como estamos viendo en https://github.com/OCA/l10n-spain/pull/232, dejarlo en un módulo separado va a facilitar un mantenimiento/migración más escalado, por lo que mi apuesta es por hacer esos cambios en el módulo _l10n_es_igic_, no en el principal.

syci commented:
Ok, entonces hacer lo mismo que l10n_es_irnr pero con el IGIC? 

> El 28 oct 2015, a las 19:46, Pedro M. Baeza notifications@github.com escribió:
> 
> No, Mario, no he dicho de juntarlo en el l10n_es, si no de crear un objeto plan contable distinto, que herede del ya existente, pero que permita seleccionar "PGCE PYMEs" y "PGCE PYMEs Canarias" por separado. Tal como estamos viendo en #232 https://github.com/OCA/l10n-spain/pull/232, dejarlo en un módulo separado va a facilitar un mantenimiento/migración más escalado, por lo que mi apuesta es por hacer esos cambios en el módulo l10n_es_igic, no en el principal.
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/OCA/l10n-spain/pull/235#issuecomment-151969273.

pedrobaeza commented:
Eso es, pero modificando los datos tal como expongo en los otros puntos.

antespi commented:
Estoy de acuerdo con @pedrobaeza en separarlo en un plan contable nuevo, es lo más mantenible. Aunque no entiendo cómo se puede hacer lo que dices en el último punto de cambiar las posiciones fiscales. Las posiciones se definen en el plan común, ¿cómo puede un plan hijo cambiar una posición fiscal del padre?

pedrobaeza commented:
Efectivamente, lo de las posiciones fiscales es territorio inexplorado, porque hasta ahora no hemos tenido que hacerlo en ningún plan contable, pero como mínimo sí que se podrían crear nuevas posiciones fiscales que por ejemplo se llamen "Canarias > España", "Canarias Intracomunitario", etc que modelen las condiciones específicas de Canarias, aunque convivan con "Régimen Nacional", "Régimen Intracomunitario", etc, y ésas no se utilicen.

daviddiz commented:
Totalmente de acuerdo con los puntos que plantea Pedro.
Estoy ya en marcha con los tres primeros:
- Renombrar códigos con la nueva nomenclatura. Ya explicaba en el Readme que estaba pendiente.
- Separación de los códigos, impuestos y cuentas en tres ficheros distintos para simplificar la lectura y edición.
- Cuentas contables no desglosadas por impuesto.

Para los otros dos puntos haré lo siguiente tal y como planteais:
- Creación de nuevo plan contable "PGCE PYMEs Canarias" que herede de "PGCE PYMEs"
- Creación de nuevas posiciones contables específicas de Canarias que convivan con las existentes.

Según vaya teniendo cosas hechas las iré subiendo los próximos días.
Un saludo.

pedrobaeza commented:
Genial, David, muchas gracias por el esfuerzo.

daviddiz committed:
Separación de los códigos, impuestos y cuentas en tres ficheros
distintos para simplificar la lectura y edición.

daviddiz committed:
-Códigos de impuestos renovados
-Cada modelo en fichero separado
-Cargar IGIC en nuevo plan contable que hereda del 2008 Pymes Español
-Cuentas contables genéricas para los impuestos, no desglosadas

syci commented:
Hola @daviddiz tengo también casi todo el código hecho, he hablado con @pedrobaeza  y me comenta que lo mejor es que yo haga un PR a tu rama de OCB cuando tenga mis cambios. 
Realmente tengo todo casi hecho y habrá que combinar y revisar cuando te haga el PR a tí.

Pero tengo que probarlo en LOCAL aún y revisar que todo está OK con los asesores.

daviddiz commented:
Lo que más trabajo puede dar es combinar las descripciones y nombres de los códigos de los impuestos y los ids de los xml.
Mira la convención que he seguido y me comentas.

daviddiz commented:
Para ahorrar trabajo, cuelgo capturas de los nuevos códigos, impuestos y cuentas creados al instalar el nuevo plan contable para canarias.

Tabla de Impuestos:
![tabla de impuestos igic](https://cloud.githubusercontent.com/assets/1212657/10944875/776a0dc2-831b-11e5-85bd-90743d572048.png)

Códigos de Impuestos:
![codigos de impuestos igic](https://cloud.githubusercontent.com/assets/1212657/10944906/9d92f392-831b-11e5-8347-c3d9ab96d911.png)
![codigos de impuestos igic 2](https://cloud.githubusercontent.com/assets/1212657/10944909/a1ffc1e4-831b-11e5-83b7-8347ca16979e.png)
![codigos de impuestos igic 3](https://cloud.githubusercontent.com/assets/1212657/10944915/a7e26170-831b-11e5-8079-78e8730e00c0.png)

Impuestos:
![impuestos](https://cloud.githubusercontent.com/assets/1212657/10944919/af70483a-831b-11e5-90c0-47e22208ef6e.png)
![impuestos 2](https://cloud.githubusercontent.com/assets/1212657/10944921/b32fe782-831b-11e5-9f3d-985d8612c99d.png)

Cuentas contables:
![cuentas contables igic](https://cloud.githubusercontent.com/assets/1212657/10944927/bc2ef314-831b-11e5-8272-285f8395d0a0.png)

rafaelbn commented:
@daviddiz @syci muchas gracias, en cuando lo tengáis, estaremos encantado de revisar y probarlo.

rafaelbn commented:
Sería muy interesante que añadiérais por favor en el README el uso del módulo, es decir, cómo probarlo. Como ejemplo: https://github.com/OCA/l10n-spain/pull/234/files

rafaelbn commented:
Buenos días, me gustaría preguntar cómo está este tema, por favor. Muchas gracias.

daviddiz commented:
Estoy teniendo las reuniones finales para implantar la contabilidad en clientes canarios, y mantengo conversaciones sobre detalles del módulo con otros desarrolladores, les he planteado publicar las conversaciones aquí para hacerlas visibles a la comunidad así que próximamente tendréis noticias.
Tomo nota de la necesidad de ampliar la información del README.

pedrobaeza commented:
David, algún progreso en el PR. ¿Se puede considerar válido?

CrazyIFriT commented:
Pedro iba a revisarlo, pero no he tenido tiempo, cuestiones personales 
tambien, disculpenme

El 21/01/16 a las 12:43, Pedro M. Baeza escribió:

> David, algún progreso en el PR. ¿Se puede considerar válido?
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/OCA/l10n-spain/pull/235#issuecomment-173559197.

daviddiz commented:
Está revisado por los contables canarios del cliente, tanto los impuestos como las posiciones fiscales y cuentas contables. Lo han empezado a utilizar ya.
Eso no quiere decir que pueda hacer alguna mejora a corto plazo cómo mejorar el README como me proponen o mejorar la instalación cuando ya existe un plan contable instalado, pero es funcional y válido tal y cómo está.
Un saludo.

pedrobaeza commented:
Ya he instalado el módulo en local y funciona correctamente, pero me quedan estas dudas/peticiones:
- Todos los códigos de impuestos del IGIC, supongo que se declaran en un único modelo (era el 420?). ¿Podrías agrupar esos códigos teniendo como padre uno que ponga el nombre del modelo tal como hacemos con el 303, 111, 115...?
- Los números entre corchetes de los códigos de impuesto se corresponden con las casillas del modelo, ¿verdad?
- Las ventas de Canarias a la Península leo que tienen un tratamiento especial (http://www.grupobelmar.onored.com/news/Year_2014/News_20140212.html), por lo que habría que crear una posición fiscal que se llame por ejemplo "Ventas a Península" que lo trate.
- Las ventas de Canarias al resto de Europa, al resto del mundo y el sujeto pasivo, ¿cómo se tratan? Si hay que hacer mapeados del IGIC al impuesto que corresponda, se tendrán que crear las posiciones fiscales correspondientes.
- ¿No existen retenciones a profesionales en Canarias?
- En la plantilla del plan contable, has puesto todos los impuestos por defecto. Sólo hay que poner los 2 del 7% (de venta y de compra).

@rafaelbn, para tu conocimiento.

CrazyIFriT commented:
Respondiendo...

El 22/01/16 a las 03:28, Pedro M. Baeza escribió:

> Ya he instalado el módulo en local y funciona correctamente, pero me 
> quedan estas dudas/peticiones:
> - Todos los códigos de impuestos del IGIC, supongo que se declaran
>   en un único modelo (era el 420?). ¿Podrías agrupar esos códigos
>   teniendo como padre uno que ponga el nombre del modelo tal como
>   hacemos con el 303, 111, 115...?
> 
> _Liquidacion de IGIC 425, cuando se es autonomo 420 estimacion directo y 
> 421 para modulos._
> - Los números entre corchetes de los códigos de impuesto se
>   corresponden con las casillas del modelo, ¿verdad?
> - Las ventas de Canarias a la Península leo que tienen un
>   tratamiento especial
>   (http://www.grupobelmar.onored.com/news/Year_2014/News_20140212.html),
>   por lo que habría que crear una posición fiscal que se llame por
>   ejemplo "Ventas a Península" que lo trate.
> 
> *Depende del tipo del servicio, si la sede esta aqui en Canarias tiene 
> que hacerlo con IGIC, sino no. Si se le cobra el IGIC tendrían que 
> devolverselo. *
> - Las ventas de Canarias al resto de Europa, al resto del mundo y el
>   sujeto pasivo, ¿cómo se tratan? Si hay que hacer mapeados del IGIC
>   al impuesto que corresponda, se tendrán que crear las posiciones
>   fiscales correspondientes.
> 
> _Sip,  el Impuesto General Indirecto Canario y el Arbitrio sobre 
> Importaciones y Entregas de Mercancías en Canarias._
> - ¿No existen retenciones a profesionales en Canarias? _Si, ahroa
>   creo que han bajado al 7_
> - En la plantilla del plan contable, has puesto todos los impuestos
>   por defecto. Sólo hay que poner los 2 del 7% (de venta y de
>   compra). _Bueno creo que depende de la actividad. el 3% están
>   poniendo en algunas empresas en caso que haya objetos metalicos de
>   por medio o algo asi, el 7% de lo general. Conozco otros que ponen
>   el 3% por productos que contengan plastico o 0% si es especial._
> 
> @rafaelbn https://github.com/rafaelbn, para tu conocimiento.
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/OCA/l10n-spain/pull/235#issuecomment-173794284.

pedrobaeza commented:
Según lo que comenta entonces @CrazyIFriT, habría que hacer algunas cosas más, @daviddiz...

pedrobaeza commented:
¿Alguna novedad sobre este PR?

CrazyIFriT commented:
La utilización del IGIC que lo probé funciona correctamente como dijo David. Lo único que queda es implementar los demás.

pedrobaeza commented:
Ya, pero es incompleto, porque hay demasiadas casuísticas que no cubre.

rafaelbn commented:
¿Podemos aceptar este y añadir un ROADMAP/KNOWN ISSUES para ir incorporando mejoras?

pedrobaeza commented:
No, yo lo considero incompleto, ya que no cubre las posiciones fiscales que ya existen en el plan PYME, y puede dar lugar a equívocos.

daviddiz commented:
En resumen, tal y como lo veo dividiré lo que falta por hacer en:
Tareas técnicas que acometeré en breve:
- Agrupar códigos teniendo como padre uno que ponga el nombre del modelo tal como hacemos con el 303, 111, 115.. En cuanto a la creación de modelos es una tarea comprometida con el cliente a medio plazo pero que se implementaré en módulos independientes.
  - Los números entre corchetes de los códigos de impuesto se corresponden con las casillas del modelo

En cuanto a cuestiones contables como la creación de nuevas posiciones fiscales y mapeo dependiendo de si son servicios o mercancías a extracomunitarios o peninsulares
- con los contables hemos llegado a la conclusión de que no es necesario crear ninguna posición fiscal nueva pero si que hay que añadir el mapeo de iva a igic en las posiciones fiscales (como en la Régimen Extracomunitario / Canarias, Ceuta y Melilla) según sean servicios o exportación de mercancías.

De todas formas acepto consejos de todo aquel que conozca la operativa de impuestos en canarias.
Iré subiendo lo que haga para que lo reviséis.

pedrobaeza commented:
Efectivamente, tu resumen me parece perfecto, y esas dos tareas serían las principales.

Las posiciones fiscales es hacerlas consecuentes para una visión desde Canarias. Posibles enfoques:
- Se sigue considerando el IVA 21% como por defecto, y se mapea en la posición fiscal "Régimen Extracomunitario / Canarias, Ceuta y Melilla" el IGIC. Ésta casi que la descarto, ya que una empresa con sede fiscal en Canarias parecería raro trabajar así.
- Se considera el IGIC 7% como por defecto, y entonces se crea una posición fiscal "Ventas a la Península", o bien se añade en el "Régimen Nacional" el mapeo de los IGIC a lo que corresponda cuando vendes a España, aunque el nombre no sea afinado del todo. Igualmente, el nombre de "Régimen Extracomunitario / Canarias, Ceuta y Melilla" no sería afinado del todo para indicar las ventas fuera de Canarias y que no son la Península, pero es lo que más se acerca, y ahí podrías mapear el IGIC a lo que corresponda. Ahora bien, si esas ventas tienen un tratamiento distinto, no se podrían agrupar. Y al contrario, si da lo mismo fiscalmente vender a la Península, a Ceuta, Melilla o extra-comunitario, entonces se puede agrupar todo en una misma posición fiscal (aunque las otras te sobren). Las ventas a la UE supongo que se tratarán parecidas a la Península, pero todo dependerá de cómo tengáis que hacerla constar en vuestro modelo de impuestos.

syci commented:
La instalación es correcta, en mi caso del uso del IGIC uso este desglose para poder separar cada tipo y poder hacer las declaraciones del 420, que al rellenar nos piden los importes por cada tipo.

Estas son las numeraciones que se usan en Canarias para desglose de os tipos:

 472000     Hacienda Pública. IVA soportado
472700  Hacienda Pública. IGIC soportado 0%
472702  Hacienda Pública. IGIC soportado 2%
472703  Hacienda Pública. IGIC soportado 3%
472705  Hacienda Pública. IGIC soportado 5%
472706  Hacienda Pública. IGIC soportado 7%
472709  Hacienda Pública. IGIC soportado 9%
472713  Hacienda Pública. IGIC soportado 13%
472735  Hacienda Pública. IGIC soportado 13,5%
472795  Hacienda Pública. IGIC soportado 9,5% 

Cuando facturo algo con IGIC 9% por ejemplo, el asiento debe de generar hacía la 472709.

pedrobaeza commented:
Mario, ese concepto es erróneo y parcial. Erróneo porque los impuestos no se controlan por cuentas contables, y parcial porque ya hemos dicho atrás las cosas que faltan.

pedrobaeza commented:
@daviddiz, alguna novedad con esto?

ezanardi commented:
En la tabla de impuestos de IVA se distingue entre bienes de inversión, bienes corrientes y servicios para el IVA soportado, y entre bienes y servicios para el repercutido. 
Así, tenemos:
...
10% IVA Soportado (bienes de inversión)    ￼ P_IVA10_BI
10% IVA soportado (bienes corrientes)   ￼ P_IVA10_BC
10% IVA soportado (servicios corrientes)    ￼ P_IVA10_SC
IVA 10% (Bienes)    ￼ S_IVA10B
IVA 10% (Servicios) ￼ S_IVA10S
...
¿No deberíamos hacer lo mismo para el IGIC?￼

pedrobaeza commented:
No si la fiscalidad de Canarias no obliga a distinguirlos. Al final, los impuestos es una forma de clasificar la información para así cumplir con las obligaciones fiscales.

Y pensar que hay otros planes contables que tienen sólo 2 impuestos de compra y dos de venta...

CrazyIFriT commented:
Si, es exactamente igual que el IVA.

El 25/05/16 a las 16:30, Pedro M. Baeza escribió:

> No si la fiscalidad de Canarias no obliga a distinguirlos. Al final, 
> los impuestos es una forma de clasificar la información para así 
> cumplir con las obligaciones fiscales.
> 
> Y pensar que hay otros planes contables que tienen sólo 2 impuestos de 
> compra y dos de venta...
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub 
> https://github.com/OCA/l10n-spain/pull/235#issuecomment-221612327

ezanardi commented:
Le he mandado a @daviddiz un PR [#1](https://github.com/daviddiz/igic/issues/1) con los cambios que indicaba @pedrobaeza el pasado 25 de Febrero. 
- Agrupar códigos de impuestos teniendo como padre uno que ponga el nombre del modelo (420)
- Se considera el IGIC 7% como por defecto, así que creamos una posición fiscal "Regimen de Importaciones/Exportaciones a/desde Canarias", que agrupa las operaciones a la Península, a Ceuta, Melilla, resto de la UE o extra-comunitario, ya que en la fiscalidad de Canarias se tratan igual.

A ver si tiene ocasión de revisarlos y añadirlos a su rama, para poder mover hacia adelante éste PR.

daviddiz commented:
Esta semana me es imposible, pero la semana que viene revisaré los cambios
propuestos y los subiré

El 9 de noviembre de 2016, 14:38, ezanardi notifications@github.com
escribió:

> Le he mandado a @daviddiz https://github.com/daviddiz un PR con los
> cambios que indicaba @pedrobaeza https://github.com/pedrobaeza el
> pasado 25 de Febrero.
> - Agrupar códigos de impuestos teniendo como padre uno que ponga el
>   nombre del modelo (420)
> - Se considera el IGIC 7% como por defecto, así que creamos una
>   posición fiscal "Regimen de Importaciones/Exportaciones a/desde Canarias",
>   que agrupa las operaciones a la Península, a Ceuta, Melilla, resto de la UE
>   o extra-comunitario, ya que en la fiscalidad de Canarias se tratan igual.
> 
> A ver si tiene ocasión de revisarlos y añadirlos a su rama, para poder
> mover hacia adelante éste PR.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/OCA/l10n-spain/pull/235#issuecomment-259416287, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/ABKA8eQOeATFElm7YUtbAKMQDj64-wezks5q8czLgaJpZM4GXUss
> .

## 

_David Diz Martínez_
Desarrollador y Analista de Software

_daviddiz@gmail.com daviddiz@gmail.com_

_amorodoo.com http://amorodoo.comImplantación y desarrollo de OdooERP
Software Libre_C/Arzobispo Gelmírez 15 bajo
15002 A Coruña
Teléfono: +34 881 899 137

CrazyIFriT commented:
@ezanardi mi opinion es que rara vez se utiliza la cuenta de desajuste de IGIC.

ezanardi commented:
@CrazyIFriT me comentaron los contables de añadirlas "por completar", pero tampoco veo que se usen mucho. ¿Ves algún inconveniente en que estén dadas de alta, por si a alguien le resultan útiles?

CrazyIFriT commented:
@ezanardi No, las puedes dejar, no está de más tenerla, pero bueno rara vez lo veo. ¿Que estás desde la peninsula o Canarias? Normalmente se suele usar una cuenta creadas personal, pero bueno es cuestion de cada empresa.

ezanardi commented:
@CrazyIFriT Estoy (estamos, Atlantux Consultores) en Canarias. En Santa Cruz de Tenerife.

ezanardi commented:
David ya ha añadido el PR [#1](https://github.com/daviddiz/igic/pull/1) a su repo, pero no se ha actualizado éste PR. Supongo que se debe a que el repo actual de David ( https://github.com/daviddiz/igic ) no es el mismo que usó para enviar éste PR. De hecho, me temo que en algún momento se ha eliminado ese repo, por que en la cabecera de éste PR aparece como "unknown repository". Así que, ¿cómo hacemos para continuar?

ezanardi commented:
Se me ocurre que lo más fácil sería crear un nuevo PR desde mi repo o el de David (ahora mismo están idénticos), haciendo referencia a éste PR, y luego cerrar éste haciendo referencia al nuevo. ¿Cómo lo ven?

CrazyIFriT commented:
Perfecto, a ver si vamos moviendo esto.

Saludos desde la isla de al lado, GC

pedrobaeza commented:
Se podría utilizar este mismo PR si se sube con el mismo nombre de rama, pero si no, lo más rápido es cerrar este PR y abrir uno nuevo.

ezanardi commented:
No tengo claro cómo utilizar el mismo nombre de rama, así que he creado el PR [#398](https://github.com/OCA/l10n-spain/pull/398).

pedrobaeza commented:
Cierro entonces ésta.

ezanardi commented:
Cierro el PR #398 y abro el #400 para que aparezca correctamente la autoría de los commits

ruslanguns commented:
Espero con esperanza que alguien noble lo prepare para la versión 11 ?

ruslanguns commented:
He intentado instalar el https://github.com/daviddiz/igic/tree/master/l10n_es_igic en un entorno Odoo:10 

Pero obtengo el error al intentar instalarlo...

Aquí un video mostrando la reproducción, con suerte es algo tonto y me dan una pista de cómo solucionarlo.

Unsaludo
Ruslan

http://www.youtube.com/watch?v=15Zes35__YE

```
Odoo Server Error

Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/odoo/http.py", line 642, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/usr/lib/python2.7/dist-packages/odoo/http.py", line 684, in dispatch
    result = self._call_function(**self.params)
  File "/usr/lib/python2.7/dist-packages/odoo/http.py", line 334, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/odoo/service/model.py", line 101, in wrapper
    return f(dbname, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/odoo/http.py", line 327, in checked_call
    result = self.endpoint(*a, **kw)
  File "/usr/lib/python2.7/dist-packages/odoo/http.py", line 942, in __call__
    return self.method(*args, **kw)
  File "/usr/lib/python2.7/dist-packages/odoo/http.py", line 507, in response_wrap
    response = f(*args, **kw)
  File "/usr/lib/python2.7/dist-packages/odoo/addons/web/controllers/main.py", line 896, in call_button
    action = self._call_kw(model, method, args, {})
  File "/usr/lib/python2.7/dist-packages/odoo/addons/web/controllers/main.py", line 884, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/usr/lib/python2.7/dist-packages/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/usr/lib/python2.7/dist-packages/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/odoo/addons/base/module/module.py", line 411, in button_immediate_install
    return self._button_immediate_function(type(self).button_install)
  File "/usr/lib/python2.7/dist-packages/odoo/addons/base/module/module.py", line 485, in _button_immediate_function
    modules.registry.Registry.new(self._cr.dbname, update_module=True)
  File "/usr/lib/python2.7/dist-packages/odoo/modules/registry.py", line 83, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/usr/lib/python2.7/dist-packages/odoo/modules/loading.py", line 377, in load_modules
    loaded_modules, update_module, models_to_check)
  File "/usr/lib/python2.7/dist-packages/odoo/modules/loading.py", line 270, in load_marked_modules
    perform_checks=perform_checks, models_to_check=models_to_check
  File "/usr/lib/python2.7/dist-packages/odoo/modules/loading.py", line 182, in load_module_graph
    _load_data(cr, module_name, idref, mode, kind='data')
  File "/usr/lib/python2.7/dist-packages/odoo/modules/loading.py", line 96, in _load_data
    tools.convert_file(cr, module_name, filename, idref, mode, noupdate, kind, report)
  File "/usr/lib/python2.7/dist-packages/odoo/tools/convert.py", line 845, in convert_file
    convert_xml_import(cr, module, fp, idref, mode, noupdate, report)
  File "/usr/lib/python2.7/dist-packages/odoo/tools/convert.py", line 915, in convert_xml_import
    obj.parse(doc.getroot(), mode=mode)
  File "/usr/lib/python2.7/dist-packages/odoo/tools/convert.py", line 796, in parse
    self.parse(rec, mode)
  File "/usr/lib/python2.7/dist-packages/odoo/tools/convert.py", line 799, in parse
    self._tags[rec.tag](rec, de, mode=mode)
  File "/usr/lib/python2.7/dist-packages/odoo/tools/convert.py", line 695, in _tag_record
    f_val = self.id_get(f_ref)
  File "/usr/lib/python2.7/dist-packages/odoo/tools/convert.py", line 782, in id_get
    res = self.model_id_get(id_str, raise_if_not_found)
  File "/usr/lib/python2.7/dist-packages/odoo/tools/convert.py", line 788, in model_id_get
    return self.env['ir.model.data'].xmlid_to_res_model_res_id(id_str, raise_if_not_found=raise_if_not_found)
  File "/usr/lib/python2.7/dist-packages/odoo/addons/base/ir/ir_model.py", line 1083, in xmlid_to_res_model_res_id
    return self.xmlid_lookup(xmlid)[1:3]
  File "<string>", line 2, in xmlid_lookup
  File "/usr/lib/python2.7/dist-packages/odoo/tools/cache.py", line 87, in lookup
    value = d[key] = self.method(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/odoo/addons/base/ir/ir_model.py", line 1072, in xmlid_lookup
    raise ValueError('External ID not found in the system: %s' % xmlid)
ParseError: "External ID not found in the system: l10n_es.account_chart_template_pymes" while parsing /mnt/extra-addons/l10n_es_igic/data/account_chart_template_igic.xml:7, near
<record id="account_chart_template_pymes_canarias" model="account.chart.template">
            <field name="name">PGCE PYMEs 2008 Canarias</field>
            <field name="complete_tax_set" eval="True"/>
            <field name="parent_id" ref="l10n_es.account_chart_template_pymes"/>
        </record>
```

fuentes73 commented:
Hola @ruslanguns 

El módulo que estas intentando instalar es para la versión [8.0](https://github.com/daviddiz/igic/blob/master/l10n_es_igic/__openerp__.py#L24) así que es normal que no funcione. 
De todas formas, el módulo no pertenece a este repositorio así que deberías abrir un issue en el repositorio al que pertenece.
